hatchery/nextflow.go

```diff
@@ -1075,11 +1083,53 @@ func setupSubnet(subnetName string, cidr string, vpcid string, ec2Svc *ec2.EC2)
 		return exsubnet.Subnets[0].SubnetId, nil
 	}
 
+	// Fetch all availability zones
+	// this is being limited to a region by the ec2svc that gets passed in.
+	describeZonesOutput, err := ec2Svc.DescribeAvailabilityZones(&ec2.DescribeAvailabilityZonesInput{})
+	if err != nil {
+		return nil, fmt.Errorf("failed to describe availability zones: %v", err)
+	}
+
+	// Make sure the selected AZ has the instance type from nextflow configuration available.
+	var selectedZone string
+	for _, zone := range describeZonesOutput.AvailabilityZones {
+		if *zone.State == "available" {
+			input := &ec2.DescribeInstanceTypeOfferingsInput{
+				LocationType: aws.String("availability-zone"),
+				Filters: []*ec2.Filter{
+					{
+						Name:   aws.String("location"),
+						Values: []*string{aws.String(*zone.ZoneName)},
+					},
+					{
+						Name: aws.String("instance-type"),
+						// TODO: Should this be configurable?
+						Values: []*string{aws.String(instanceType)},
+					},
+				},
+			}
+			result, err := ec2Svc.DescribeInstanceTypeOfferings(input)
+			if err != nil {
+				return nil, fmt.Errorf("Error describing instance type offerings: %v", err)
+			}
+			if len(result.InstanceTypeOfferings) > 0 {
+				Config.Logger.Printf("Debug: Zone: %v has instance type g4dn.xlarge available. Using that for subnet", *zone.ZoneName)
```

Comment:
Maybe change this from the hardcoded `g4dn.xlarge` to `{instanceType}` variable.